llvm/lib/Transforms/Vectorize/SandboxVectorizer/SeedCollector.cpp

```diff
@@ -131,4 +141,61 @@ void SeedContainer::print(raw_ostream &OS) const {
 LLVM_DUMP_METHOD void SeedContainer::dump() const { print(dbgs()); }
 #endif // NDEBUG
 
+template <typename LoadOrStoreT> static bool isValidMemSeed(LoadOrStoreT *LSI) {
```

Comment:
a lot of the checks here don't seem to be tested

Reply:
It's always hard to decide where to checkpoint work in progress.

From the original description:

"There are many more tests to add, but I would like to get this reviewed
and the details sorted out before it grows too big."

Tests coming are for loads and additional validity checks.